containerd-shim-spin-v1/src/main.rs

```diff
@@ -42,13 +45,55 @@ pub fn prepare_module(bundle: String) -> Result<(PathBuf, PathBuf), Error> {
     // if stripped.is_some() {
     //     cmd = stripped.unwrap().to_string();
     // }
-    let working_dir = oci::get_root(&spec)?;
+    let working_dir = oci::get_root(&spec);
     let mod_path = working_dir.join("spin.toml");
     Ok((working_dir.to_path_buf(), mod_path))
 }
 
+impl Wasi {
+    async fn build_spin_application(
+        mod_path: PathBuf,
+        working_dir: PathBuf,
+    ) -> Result<spin_manifest::Application, Error> {
+        Ok(spin_loader::from_file(mod_path, working_dir).await?)
+    }
+
+    async fn build_spin_trigger(
+        engine: spin_engine::Engine,
```

Comment:
If I understand correctly, the `spin_engine::Engine` is currently just a wrapper around the `wasmtime::Engine`~ right? (source: [here](https://github.com/fermyon/spin/blob/1288afce34e158720a13864e42b9c302b6909046/crates/engine/src/lib.rs#L66))

Reply:
yes and it implements a `inner()` function to get the engine from the wrapper, but there is no method to `set` the engine.

Reply:
Awesome~AAAND... Is it fair to assume that this method exists because, in the future, they want to support other engines?